docs/docs/user/references/components.md

```diff
@@ -0,0 +1,245 @@
+# Components
+
+This section explains the different components of KPOps, their usage and
+configuration. To learn more about their hierarchy, visit the
+[architecture](/docs/developer/architecture/component-inheritance.md) page.
+
+## KubernetesApp
+
+### Usage
+
+Can be used to deploy any app in Kubernetes using Helm, for example, a REST
+service that serves Kafka data.
+
+### Configuration
+
+??? "`pipeline.yaml`"
+
+    ```yaml
+      --8<--
+      ./docs/resources/pipeline-components/kubernetes-app.yaml
+      --8<--
+    ```
```

Comment:
It is empty:
<img width="714" alt="image" src="https://user-images.githubusercontent.com/20357405/220644313-933f2572-482f-4bfc-9b9d-bb8bf76c1975.png">

Reply:
Strange, did you do `poetry install --with docs`?

Reply:
hmm I actually ran it because my environment didn't contain mkdocs. But didn't work for me. Does this need an extension or package?
How about @disrupted and @philipp94831?

Reply:
It relies on ` pymdownx.snippets` which was already in `mkdocs.yaml`

Reply:
for me it's also empty

Reply:
Hmm how do you render it?

Reply:
When I use `just serve-docs` it is in fact empty, when I use `mkdocs serve`, it works as intended. It has to do with the path I think, since I'm giving a relative path for the config examples

Reply:
fixed, had to set a base path for the snippet extension

Reply:
looks great!